### PR TITLE
Fix to remove webgl warnings resulted from blank texts in scatter3d

### PR DESCRIPTION
**Pull request by archmoj**
This PR resolves issue #704.
@alexcjohnson

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -80,7 +80,7 @@
     "gl-plot2d": "^1.3.1",
     "gl-plot3d": "^1.5.7",
     "gl-pointcloud2d": "^1.0.1",
-    "gl-scatter3d": "^1.0.11",
+    "gl-scatter3d": "git://github.com/gl-vis/gl-scatter3d.git#2ee9fbb3c29af694ad64258a27dc04d617896e84",
```
Comment (archmoj) [inline package.json]: https://github.com/gl-vis/gl-scatter3d/compare/master...issue-704

Comment (alexcjohnson) [reply package.json]: BTW now that you have PRs in these other repos, even better than the branch link would be to link to that PR directly
https://github.com/gl-vis/gl-scatter3d/pull/9

**Comment by alexcjohnson** [comment]
Are the warnings that this PR suppresses accessible to us in a jasmine test? Or is there some other effect of the error that we *can* test?

File: package.json (hunk 1 above)
Comment (archmoj) [reply package.json]: Thanks for the note. But the other PR on the gl-scatter3d repo is not opened by me.

**Comment by etpinard** [comment]
> Are the warnings that this PR suppresses accessible to us in a jasmine test?

Unfortunately no.

> Or is there some other effect of the error that we can test?

`gd._fullLayout.scene._scene.glplot.objects[0].glyphBuffer.length` was 0 before and is `2616` after plotting:

```js
Plotly.newPlot(gd, [{
  type: 'scatter3d',
  mode: 'text',
  x: [1,2,3],
  y: [2,1,2],
  z: [2,1,3]
}])
```

@archmoj is this expected? Could we use it to :lock: down this fix?

Also referencing https://github.com/gl-vis/gl-scatter3d/pull/9/files#r227874673

**Comment by archmoj** [comment]
Thanks @etpinard.
Yes this is really helpful.

**Comment by etpinard** [comment]
- append an `it` to this [describe block](https://github.com/plotly/plotly.js/blob/master/test/jasmine/tests/gl3d_plot_interact_test.js#L24)
- call

```js
Plotly.plot(gd, [{
  type: 'scatter3d',
  mode: 'text',
  x: [1,2,3],
  y: [2,1,2],
  z: [2,1,3]
}])
```

- assert that `gd._fullLayout.scene._scene.glplot.objects[0].glyphBuffer.length` isn't 0

- add a comment above that asserting linking to issue https://github.com/plotly/plotly.js/issues/704

File: package.json (hunk 1 above)
Comment (archmoj) [reply package.json]: My apologies @alexcjohnson, 
Here is the link to the open PR:
https://github.com/gl-vis/gl-scatter3d/pull/9

**Comment by archmoj** [comment]
The jasmine test is now able to catch the bug. Many thanks @etpinard for the helpful introduction to this test.
Next step: I need to finalize the changes to gl-scatter3d module, and I may try to fix this issue as well.
https://github.com/plotly/plotly.js/issues/1976

**Comment by archmoj** [comment]
Regarding this issue and other issues I noticed in gl-scatter3d module please refer to [this demo](https://codepen.io/MojtabaSamimi/pen/OBrdyE).
To test improvements and new functionalities (e.g. handling JSON data in the text) with the new version please run the JavaScript code (excluding its first line) directly in the console.

**Comment by etpinard** [comment]
Nice demo @archmoj 

Have you check that those non-string items in `text` behave the same way in for `scatter` traces:

https://codepen.io/etpinard/pen/qJgPvX ?

**Comment by archmoj** [comment]
Thank you @etpinard for this demonstration. Good call if needed I could also fix the other trace in order to e.g. display JSON object properly?

**Comment by etpinard** [comment]
> I could also fix the other trace in order to e.g. display JSON object properly?

No need. I don't think we should support that. We could do better than rendering `[object object]` but that'll be a another day.

**Comment by archmoj** [comment]
So now on scatter3d we render `undefined` as empty text. Sounds OK? @etpinard
Using your [demo for scatter](https://codepen.io/etpinard/pen/qJgPvX ) one can see they are rendered as blank which is similar to the new scatter3d behavior. 
Please note that we used to render them as `undefined` which could have been a bit long & annoying.
Any other suggestion? @alexcjohnson

**Comment by alexcjohnson** [comment]
I'd call it a bug if `undefined` (or `null`) *ever* gets drawn as something other than a missing / blank value. No need to hunt around and fix other trace types here, but it's definitely a positive change for `scatter3d`.

**Comment by etpinard** [comment]
> Sounds OK?

Looks good! Could you add JSON version of your https://codepen.io/MojtabaSamimi/pen/OBrdyE to our test mocks?

**Comment by archmoj** [comment]
Great idea @etpinard . 
Yes I'll do it right now.

**Comment by archmoj** [comment]
There is a question. JSON format only has `null` and does not have `undefined`. So how we could/should handle cases where the user used `undefined` and of course not "undefined" (as string) there?
Should we those cases to `null` or `""` (blank) or  `"undefined"` (string)? I prefer not to use `"undefined"` as string in this scenario. Your thoughts? @etpinard @alexcjohnson

**Comment by etpinard** [comment]
> There is a question. JSON format only has `null` and does not have `undefined`. So how we could/should handle cases where the user used `undefined` and of course not "undefined" (as string) there?
> Should we those cases to `null` or `""` (blank) or `"undefined"` (string)? I prefer not to use `"undefined"` as string in this scenario.

`null`, `undefined`, `''` should all result in a blank text entry.

**Comment by archmoj** [comment]
Yes, but it is not clear to me how one could mimic the case of `undefined` using JSON input? 
One idea could be to write nothing between two commas. Right?

**Comment by etpinard** [comment]
Pro tip @archmoj you can go ahead and close "duplicate" PRs. If something goes wrong, we can always reopen closed PRs.

**Comment by archmoj** [comment]
Thanks for the note @etpinard. OK I would follow that next time.

**Pull request closed without merging**